vector/src/main/java/im/vector/riotx/features/home/room/list/RoomListViewModel.kt

```diff
@@ -231,21 +227,13 @@ class RoomListViewModel @Inject constructor(initialState: RoomListViewState,
                     }
                 }
 
-        val roomComparator = when (displayMode) {
-            RoomListFragment.DisplayMode.HOME     -> chronologicalRoomComparator
-            RoomListFragment.DisplayMode.PEOPLE   -> chronologicalRoomComparator
-            RoomListFragment.DisplayMode.ROOMS    -> chronologicalRoomComparator
-            RoomListFragment.DisplayMode.FILTERED -> chronologicalRoomComparator
-            RoomListFragment.DisplayMode.SHARE    -> chronologicalRoomComparator
-        }
-
         return RoomSummaries().apply {
-            put(RoomCategory.INVITE, invites.sortedWith(roomComparator))
-            put(RoomCategory.FAVOURITE, favourites.sortedWith(roomComparator))
-            put(RoomCategory.DIRECT, directChats.sortedWith(roomComparator))
-            put(RoomCategory.GROUP, groupRooms.sortedWith(roomComparator))
-            put(RoomCategory.LOW_PRIORITY, lowPriorities.sortedWith(roomComparator))
-            put(RoomCategory.SERVER_NOTICE, serverNotices.sortedWith(roomComparator))
+            put(RoomCategory.INVITE, invites)
+            put(RoomCategory.FAVOURITE, favourites)
+            put(RoomCategory.DIRECT, directChats)
+            put(RoomCategory.GROUP, groupRooms)
+            put(RoomCategory.LOW_PRIORITY, lowPriorities)
+            put(RoomCategory.SERVER_NOTICE, serverNotices)
```

Comment:
No more sorting here?

Reply:
Sorting is made before now, as we are currently using the same comparator for every sections